corral/log/log.pony

```diff
@@ -0,0 +1,64 @@
+"""
+# Logging in corral
+
+Logging in corral uses the 'logger' package from the standard library, with
+custom implementations of the 'LogFormatter' interface.
+
+The API for 'logger' makes use of boolean short-circuiting in order to avoid the
+work of building strings that won't get logged. This can make log statements
+somewhat verbose:
+
+```pony
+let logger = StringLogger(Warn, env.out)
+logger(Info) and logger.log("this message won't get" + " built or" + "logged")
+logger(Error) and logger.log("this message will")
+```
+
+The boilerplate involved in logging might prompt someone to write some 'Log'
+wrapper class with methods like `info()` and `warn()` that check against a field
+that stores the log level. The drawback of this approach is that it cannot
+benefit from the work-saving that boolean short-circuiting provides, because
+any message argument passed to a method in this 'Log' class would have to be
+built.
+"""
+
+use "logger"
+
+primitive LevelLogFormatter
+  """
+  LevelLogFormatter produces LogFormatters that prepend a log level to log
+  messages. Log messages that contain the log level are easier to search and to
+  filter.
+
+  LevelLogFormatter avoids the need to prepend the log level at all logging call
+  sites by creating a special-purpose LogFormatter whose sole purpose is to
+  prepend the log level.
+
+  LevelLogFormatter is used by the logger(s) created in Main.
+  """
+  fun apply(level: LogLevel): LogFormatter =>
+    """
```

Comment:
Lines 33 to 35 should be most of the apply method comment.

Creates lambda to... blah blah blah.